api/lib/logger.ts

```diff
@@ -177,9 +185,153 @@ export function module(module: string): ModuleLogger {
  * Setup all loggers starting from config
  */
 export function setupAll(config: DeepPartial<GatewayConfig>) {
+	if (cleanJob) {
+		clearInterval(cleanJob)
+		cleanJob = undefined
+	}
+
 	logContainer.loggers.forEach((logger: ModuleLogger) => {
 		logger.setup(config)
 	})
 }
 
+let cleanJob: NodeJS.Timeout
+
+function setupCleanJob(settings: DailyRotateFileTransportOptions) {
+	if (cleanJob) {
+		return
+	}
+
+	let maxFilesMs: number
+	let maxFiles: number
+	let maxSizeBytes: number
+
+	const logger = module('LOGGER')
+
+	// convert maxFiles to milliseconds
+	if (settings.maxFiles !== undefined) {
+		const matches = settings.maxFiles.toString().match(/(\d+)([dhm])/)
+
+		if (settings.maxFiles) {
+			const value = parseInt(matches[1])
+			const unit = matches[2]
+			switch (unit) {
+				case 'd':
+					maxFilesMs = value * 24 * 60 * 60 * 1000
+					break
+				case 'h':
+					maxFilesMs = value * 60 * 60 * 1000
+					break
+				case 'm':
+					maxFilesMs = value * 60 * 1000
+					break
+			}
+		} else {
+			maxFiles = Number(settings.maxFiles)
+		}
+	}
+
+	if (settings.maxSize !== undefined) {
+		// convert maxSize to bytes
+		const matches2 = settings.maxSize.toString().match(/(\d+)([kmg])/)
+		if (matches2) {
+			const value = parseInt(matches2[1])
+			const unit = matches2[2]
+			switch (unit) {
+				case 'k':
+					maxSizeBytes = value * 1024
+					break
+				case 'm':
+					maxSizeBytes = value * 1024 * 1024
+					break
+				case 'g':
+					maxSizeBytes = value * 1024 * 1024 * 1024
+					break
+			}
+		} else {
+			maxSizeBytes = Number(settings.maxSize)
+		}
+	}
+
+	// clean up old log files based on maxFiles and maxSize
+
+	const filePathRegExp = new RegExp(
+		path.basename(settings.filename).replace(/%DATE%/g, '([^.]+)'),
```

Comment:
@Apollon77 in that case `[^.]+` means: take everything that is not a dot. So it will stop just before the `.log`. This of course could be broken in case `%DATE%` contains a dot but 🤷🏼‍♂️

Reply:
[z-ui_2024-02-13.log](https://regex101.com/r/kKW5ox/1)

Reply:
I created a new version of the regex to make it stronger, it will work for every date format, it just takes everything between the %DATE% prefix/suffix:

https://regex101.com/r/Z3VR1I/1